grails-app/services/au/org/ala/ecodata/ActivityService.groovy

```diff
@@ -178,6 +178,32 @@ class ActivityService {
     }
 
     /**
+     * Deletes all activities associated with project activityId.
+     *
+     * @param pActivityId project activity id
+     * @param destroy if true will really delete the object
+     * @return
+     */
+    Map deleteByProjectActivity(String pActivityId, boolean destroy = false) {
+        Map result
+
+        ProjectActivity pActivity = ProjectActivity.findByProjectActivityIdAndStatus(pActivityId, ACTIVE)
```

Comment:
Why only look for ACTIVE ProjectActivities? We are deleting, so the current state shouldn't really matter.